lamindb/models/sqlrecord.py

```diff
@@ -912,50 +929,29 @@ def save(self: T, *args, **kwargs) -> T:
                         f"returning {self.__class__.__name__.lower()} with same hash: {pre_existing_record}"
                     )
                     init_self_from_db(self, pre_existing_record)
-                elif (
-                    self.__class__.__name__ == "Storage"
-                    and isinstance(e, IntegrityError)
-                    and ("root" in error_msg or "uid" in error_msg)
-                    and unique_constraint_error_in_error_message(error_msg)
-                ):
-                    # even if uid was in the error message, we can retrieve based on
-                    # the root because it's going to be the same root
-                    pre_existing_record = self.__class__.get(root=self.root)
```

Comment:
Could you add this as an example in the comments in the new abstracted implementation? Just so that it's easy for a developer to understand what's happening in terms of examples. The new code is good except for not providing these examples.